### ISSUE TITLE
Cloud Storage: Add image/video/file storage implementation

### ISSUE THREAD

**Issue by sbis04** [sbis04/flutterfire-samples#17]
Add functionality for retrieving an image, video, or other files from the device storage and upload it to Cloud Storage. Use the generated URL to provide an option to download the file on the second page.

So, there would be two pages basically: 
* **First Page** to select and upload a file from the device
* **Second Page** to download the file to the device

> **Please try to maintain similar theming across the pages, though there aren't any concrete theming guidelines for this contribution but keep this in mind for future-proofing this contribution.**

> Use the correct folder structure for adding the functionality, have a look at the readme and the project `lib` folder structure to understand the proper folder structure to follow. Each functionality is under its own category and consists mainly of `screens`, `utils`, and `widgets` folders.

> **NOTE:** You need to create your own Firebase project and configure it before starting to contribute, make sure you read the [CONTRIBUTING](https://github.com/sbis04/flutterfire-samples/blob/main/CONTRIBUTING.md) guidelines.

> Assume the Cloud Storage rule to be open to everyone (even without auth) for this demo.

More info: https://firebase.google.com/docs/storage/flutter/start
Package to use: https://pub.dev/packages/firebase_storage

> **Hacktoberfest 2022 Note:** There's no assignment of issues (first come, first review basis), all eligible contributions will be given “hacktoberfest-accepted” label to make it count towards the Hacktoberfest contribution, even if it's not merged.

**Comment by owaistaha** [comment]
I am open to work on it

**Comment by sbis04** [comment]
Nice, you can start working on this @Engr-Owais. Looking forward to your PR!

**Comment by rktejesh** [comment]
Hey @sbis04 
Can I work on this?

**Comment by sbis04** [comment]
Yes, you can work on this @rktejesh. All issues are open to everyone.
